files/file.go

```diff
@@ -95,6 +95,37 @@ func NewFileInfo(opts FileOptions) (*FileInfo, error) {
 	return file, err
 }
 
+func NewFolderInfo(opts FileOptions) (*FileInfo, error) {
```

Comment:
I think it makes sense to add a flag to the FileOptions and use NewFileInfo instead. Similar to the Expand flag.